types/stylelint/index.d.ts

```diff
@@ -42,12 +42,14 @@ type FileCache = {
 // method declarations tricks TypeScript into bivariant signature checking. More details can
 // be found here: https://stackoverflow.com/questions/52667959/what-is-the-purpose-of-bivariancehack-in-typescript-types.
 // and in the original discussion: https://github.com/stylelint/stylelint/pull/6147#issuecomment-1155337016.
+type RuleFuncArgs = (string | number | boolean | RegExp | string[])[];
```

Comment:
If you are adding a type please put it before the comment.
i.e. the comment needs to remain where it was